rust/core/src/jsonbuilder.rs

```diff
@@ -0,0 +1,60 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+ use std::ffi::{c_char, CString};
+
+// Jsonbuilder opaque with implementation using C API to feel like usual
+#[repr(C)]
+pub struct JsonBuilder {
+    _data: [u8; 0],
+}
+
+// we do not have more context than what C API gives : a boolean that there was an error
+#[derive(Debug, PartialEq, Eq)]
+pub enum JsonError {
+    CBoolean,
+}
+
+extern "C" {
+    pub fn jb_set_string(jb: &mut JsonBuilder, key: *const c_char, val: *const c_char) -> bool;
+    pub fn jb_close(jb: &mut JsonBuilder) -> bool;
+    pub fn jb_open_object(jb: &mut JsonBuilder, key: *const c_char) -> bool;
+}
+
+impl JsonBuilder {
```

Comment:
Not in core as well. Makes it confusing what should be used from our Rust.

I've been somewhat against putting *friendly* wrappers around C APIs in our `sys` crate just because its against Rust convention. But perhaps it should be our convention? Didn't we have some other PR discussion on this somewhere?

It does however mean when you are working on our Rust code, you still have 2 possible JsonBuilder structs to choose from which isn't ideal. Not quite sure how to deal with that. But it doesn't make for a nice API in my opinion.

But at least `sys` implies some form of extern/ffi interface, where `core` implies core functionality. So its still a better place.